src/toast/tod/crosstalk.py

```diff
@@ -0,0 +1,587 @@
+            raise TypeError('name and data has to be numpy arrays')
+
+    @property
+    def names_str(self) -> 'List[str]':
+        """names in list of str."""
+        return [name.decode() for name in self.names]
+
+    @classmethod
+    def load(cls, path: 'Path', *, debug: 'bool' = False):
+        """Load from an HDF5 file."""
+        with h5py.File(path, 'r') as f:
+            names = f["names"][:]
+            data = f["data"][:]
+        return cls(names, data, debug=debug)
+
+    def dump(
+        self,
+        path: 'Path',
+        *,
+        libver: 'str' = 'latest',
+        # h5py.File.create_dataset args
+        compression: 'str' = 'gzip',
+        shuffle: 'bool' = True,
+        fletcher32: 'bool' = True,
+        track_times: 'bool' = False,
+        compression_opts: 'int' = 9,
+        **kwargs,
+    ):
+        """Dump to an HDF5 file.
+
+        `libver` will be passed to h5py.File,
+        and the rest keyword arguments will be passed to `h5py.File.create_dataset`
+        with sensible defaults.
+        """
+        with h5py.File(path, 'w', libver=libver) as f:
+            f.create_dataset(
+                'names',
+                data=self.names,
+                compression=compression,
+                shuffle=shuffle,
+                fletcher32=fletcher32,
+                track_times=track_times,
+                compression_opts=compression_opts,
+                **kwargs,
+            )
+            f.create_dataset(
+                'data',
+                data=self.data,
+                compression=compression,
+                shuffle=shuffle,
+                fletcher32=fletcher32,
+                track_times=track_times,
+                compression_opts=compression_opts,
+                **kwargs,
+            )
+
+
+class OpCrosstalk(Operator):
+    """Operator that applies crosstalk matrix to detector ToDs.
+
+    :param n_crosstalk_matrices: total no. of crosstalk matrices
+    :param crosstalk_matrices: the crosstalk matrices.
+        In MPI case, this holds only those matrices owned by a rank dictate
+        by the condition `i % PROCS == RANK`
+    :param name: this name is used to save data in tod.cache, so better be unique from other cache
+
+    In a typical scenario, the classmethod `read` is used to create an object
+    instead of initiating directly.
+    """
+
+    def __init__(
+        self,
+        n_crosstalk_matrices: 'int',
+        crosstalk_matrices: 'List[SimpleCrosstalkMatrix]',
+        *,
+        name: 'str' = "crosstalk",
+        debug: 'bool' = False,
+    ):
+        self.n_crosstalk_matrices = n_crosstalk_matrices
+        self.crosstalk_matrices = crosstalk_matrices
+        self.name = name
+        self.debug = debug
+
+    def _get_crosstalk_matrix(self, i: 'int') -> 'SimpleCrosstalkMatrix':
+        """Get the i-th crosstalk matrix, used this with MPI only.
+        """
+        debug = self.debug
+
+        rank_owner = i % PROCS
```

Comment:
The communicator should be passed as an argument to this function and used in this calculation, since it may be different from the world comm.

Reply:
See https://github.com/hpc4cmb/toast/pull/380#discussion_r663285530 & https://github.com/hpc4cmb/toast/pull/380#discussion_r664973869

(But I will move the global var local to SimpleCrosstalkMatrix instead.)